text/1440-drop-types-in-const.md

```diff
@@ -6,27 +6,27 @@
 # Summary
 [summary]: #summary
 
-Allow types with destructors to be used in `static` items and in `const` functions, as long as the destructor never needs to run in const context.
+Allow types with destructors to be used in `static` items, `const` items, and `const` functions.
 
 # Motivation
 [motivation]: #motivation
 
 Some of the collection types do not allocate any memory when constructed empty (most notably `Vec`). With the change to make leaking safe, the restriction on `static` items with destructors
 is no longer required to be a hard error (as it is safe and accepted that these destructors may never run).
 
-Allowing types with destructors to be directly used in `const` functions and stored in `static`s will remove the need to have
+Allowing types with destructors to be directly used in `const` functions and stored in `static`s or `const`s will remove the need to have
 runtime-initialisation for global variables.
 
 # Detailed design
 [design]: #detailed-design
 
-- Lift the restriction on types with destructors being used in statics.
+- Lift the restriction on types with destructors being used in `static` or `const` items.
  - `static`s containing Drop-types will not run the destructor upon program/thread exit.
+ - `const`s containing Drop-types _will_ run the destructor at the appropriate point in the program.
  - (Optionally adding a lint that warn about the possibility of resource leak)
 - Alloc instantiating structures with destructors in constant expressions,
-- Continue to prevent `const` items from holding types with destructors.
 - Allow `const fn` to return types with destructors.
-- Disallow constant expressions which would result in the destructor being called (if the code were run at runtime).
+- Disallow constant expressions resulting in destructors being called at runtime (i.e: a `drop(foo)` in a `const fn`).
```

Comment:
Oh, right, I prefer the original phrasing because it emphasizes that this is not about code ran at runtime, but rather some compile-time constant evaluation context. I'm not even sure we check this correctly with the feature flag enabled, it's a bit tricky if you can't rely on a full ban of drop types.